vmware_exporter/vmware_exporter.py

```diff
@@ -574,6 +573,37 @@ def _vmware_get_vm_perf_metric(self, content, counter_info, perf_metric,
             log("Error, cannot get vm metric {0} for {1}".format(perf_metric_name,
                                                                  vm_metadata))
 
+    def _vmware_get_vmguests(self, content, vmguest_metrics, inventory):
+        """
+        Get VM Guest information
+        """
+
+        virtual_machines = self._vmware_get_obj(content, [vim.VirtualMachine])
+        log("Total Virtual Machines: {0}".format(len(virtual_machines)))
+        for virtual_machine in virtual_machines:
+            self.threader.thread_it(self._vmware_get_vmguests_metrics,
+                                    [content, virtual_machine, vmguest_metrics, inventory])
+
+    def _vmware_get_vmguests_metrics(self, content, virtual_machine, vmguest_metrics, inventory):
+        """
+        Get VM Guest Metrics
+        """
+
+        summary = virtual_machine.summary
+
+        vm_name, vm_host_name, vm_dc_name, vm_cluster_name = self._vmware_vm_metadata(inventory, virtual_machine,
+                                                                                      summary)
+
+        # gather disk metrics
+        if len(virtual_machine.guest.disk) > 0:
```

Comment:
What do you think about moving the for loop before the metric?

Reply:
sorry, not sure I follow.

Reply:
Something like 
```
for disk in virtual_machine.guest.disk: 
     vmguest_metrics['vmware_vm_guest_disk_capacity'].add_metric([vm_name, vm_host_name, vm_dc_name, vm_cluster_name, disk.diskPath], disk.capacity)
```

for both of the metrics?

Reply:
Mmmm... I've copied this block as-is, but I think I see now what you mean, it loops over the disks twice, each time extracting a different attribute (disk.freeSpace than disk.capacity).

Indeed we could change this to just a single loop and create both metrics in each iteration, is that what you've meant?

Reply:
Exactly.

Reply:
updated